### DISCUSSION TITLE
rtsps streams as microphone

### DISCUSSION THREAD

**Discussion by huytman** [tphakala/birdnet-go#66] [General]
Using birdnet pi with audio streams from a couple of outdoor cameras - audio piped in via rtsps.  Would like to install this via docker but not sure if I use rtsps streams in place of the microphones. Is this possible (yet?)? Thanks - the web ui looks very nice. Glad to see this project moving forward,

**Comment by tphakala** [comment]
Yes, RTSP stream is supported, only one stream per BirdNET-Go instance at this time. Docker support is still very young and there is quite bit of rough edges so please don't expect perfect or even trouble free experience.

Run docker image with "realtime --rtsp url" parameter like

`docker run -ti -v /path/to/config:/config -v /path/to/data:/data ghcr.io/tphakala/birdnet-go realtime --rtsp rtsp://user:password@192.168.4.175:554`

**Comment by huytman** [comment]
This is my docker config - (I had an 8080 conflict to changed to 7979)

version: "3.3"
services:
  birdnet-go:
    tty: true
    stdin_open: true
    ports:
      - 7979:8080
    volumes:
      - /home/me/config/birdnetgo/config:/config
      - /home/me/config/birdnetgo/data:/data
    image: ghcr.io/tphakala/birdnet-go:latest
    command: realtime --rtsp rtsps://my.rtsps.stream
    runtime: nvidia
    restart: unless-stopped
networks: {}
--------------------
So far working quite well with minimal CPU usage. Hoping multiple rtsps streams are supported. But so far quite a bit more responsive than the birdnet-pi I'm running.

**Comment by lhanneus** [comment]
Does multiple rtsp stream allow us to know from which source the bird recognition comes ?

**Reply by cegan09** [reply]
Not really, no. Unless I've just missed it entirely there is no indication of the stream source anywhere in the audio files that are captured. 
BirdNET-Pi has a similar problem, but it at least will put "RTSP_1" and "RTSP_2" in the file name so you can track which it came from so long as you remember which stream you added first/second/etc.

**Reply by Lenbok** [reply]
If you are subscribing to detection events via mqtt, it includes a "Source" field that identifies which stream the detection came from. There is a sub field for a source name (but I have no idea how that gets assigned), as well as the source url - in my home assistant dashboard card I look for hostname substrings in the url and map to a human readable location, e.g.:
```
  - type: custom:mushroom-template-card
    primary: ""
    secondary: >-
      {% set sourceId =
      states.sensor.birdnet_go_sightings.attributes.Source.displayName %} {% set
      sourceName = 'big shed' if 'pi2-rtsp-1' in sourceId else 'granny flat' if
      'ipcam' in sourceId else 'house' if 'esp32-rtsp-1' in sourceId else
      'mystery location' %} Heard at the {{ sourceName }} at {{
      states.sensor.birdnet_go_sightings.attributes.Time }}
    entity: sensor.birdnet_go_sightings
    icon: ""
```
